src/components/onboarding/sign-up-oauth/index.js

```diff
@@ -0,0 +1,113 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core'
+import React from 'react'
+import ReactReduxFirebase from 'react-redux-firebase'
+import FooterButton from 'components/common/footer-button'
+import FacebookButton from 'components/common/facebook-button'
+import GoogleButton from 'components/common/google-button'
+import GithubButton from 'components/common/github-button'
+import ROUTE from 'constants/route'
+import PROVIDER from 'constants/auth-provider'
+import BackButton from 'components/common/back-button'
+import style from './index.style.js'
+
+export default class SignInOAuth extends React.Component {
+  state = {
+    refreshIntervalId: null,
+  }
+
+  handleSignInWithEmail = async () => {
+    const { history } = this.props
+    history.push(ROUTE.PATH.SIGN_UP)
+  }
+
+  handleSignInOAuth = async service => {
+    const { signInWithOAuth } = this.props
+    try {
+      signInWithOAuth(service)
+    } catch (error) {
+      console.log(error)
+    }
+  }
+
+  renderTitle = () => {
+    return (
+      <div css={style.title}>
+        <p css={style.title__text}>Create your account</p>
+      </div>
+    )
+  }
+
+  renderButtons() {
+    return <div css={style.buttons}>{this.renderContinueButton()}</div>
+  }
+
+  checkAuthStatusAndStage = () => {
+    const { userId, storeId, history } = this.props
+
+    if (ReactReduxFirebase.isLoaded()) {
+      clearInterval(this.refreshIntervalId)
+
+      if (userId && storeId) {
+        history.push(ROUTE.PATH.STORE)
+      }
+
+      if (userId && !storeId) {
+        history.push(ROUTE.PATH.SIGN_UP_STORE)
+      }
+    }
+  }
+
+  componentDidMount = () => {
+    this.refreshIntervalId = setInterval(() => {
+      this.checkAuthStatusAndStage()
+    }, 1000)
+  }
+
+  renderContinueButton = () => {
+    return (
+      <div css={style.buttons}>
+        <FooterButton onClick={this.handleSignInWithEmail}>
+          Continue with e-mail
+        </FooterButton>
+        <div>or</div>
+
+        <GoogleButton
+          onClick={() => {
+            this.handleSignInOAuth(PROVIDER.GOOGLE)
+          }}
+        >
+          <div>Sign up with Google</div>
+        </GoogleButton>
+
+        <GithubButton
+          onClick={() => {
+            this.handleSignInOAuth(PROVIDER.GITHUB)
+          }}
+        >
+          <div>Sign up with Github</div>
+        </GithubButton>
+
+        <FacebookButton
+          onClick={() => {
+            this.handleSignInOAuth(PROVIDER.FACEBOOK)
+          }}
```

Comment:
Making in-line functions for props is sometimes not performant. Since these functions are re-created on every render, the prop that they are assigned to changes and that triggers re-render of the child even for PureComponents.

Best practice is to define a handler as a class method. Not really important here, just FYI.